partition_snapshot_reader.hh

```diff
@@ -26,22 +26,46 @@
 #include "clustering_key_filter.hh"
 #include <boost/range/algorithm/heap_algorithm.hpp>
 
-template <typename Accounter>
+template <bool Reversing, typename Accounter>
 class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public Accounter {
-    struct rows_position {
-        mutation_partition::rows_type::const_iterator _position;
-        mutation_partition::rows_type::const_iterator _end;
+    template<bool> struct rows_position_;
+    template<> struct rows_position_<false> {
+        using iter_type = mutation_partition::rows_type::const_iterator;
+        iter_type _position, _end;
     };
+    template<> struct rows_position_<true> {
+        using iter_type = mutation_partition::rows_type::const_reverse_iterator;
+        // The field order is reversed on purpose.
+        iter_type _end, _position;
+    };
+    using rows_position = rows_position_<Reversing>;
```

Comment:
nit: this could be achieved with `std::condition` too, with less boilerplate.

Reply:
I considered it but given the reversed order of fields, it no longer is more succinct with std::conditional.